Sources/FuturedArchitecture/Architecture/Coordinator.swift

```diff
@@ -47,26 +49,22 @@ public extension Coordinator {
     ///   - destination: The description of the desired view passed to the ``scene(for:)`` function
     ///   of the *coordinator*.
     ///   - type: Kind of modal presentation.
+    @MainActor
     func present(modal destination: Destination, type: ModalCoverModelStyle) {
         switch type {
         case .sheet:
-            Task { @MainActor in
-                self.modalCover = .init(destination: destination, style: .sheet)
-            }
+            self.modalCover = .init(destination: destination, style: .sheet)
         #if !os(macOS)
         case .fullscreenCover:
-            Task { @MainActor in
-                self.modalCover = .init(destination: destination, style: .fullscreenCover)
-            }
+            self.modalCover = .init(destination: destination, style: .fullscreenCover)
```

Comment:
Since all "scene change" methods are invoked from `@MainActor` and write into `@MainActor` bound storage, adding the `@MainActor` isolation suppresses warnings and confirms what is already de-facto the state.